backend/src/util.rs

```diff
@@ -0,0 +1,37 @@
+use load::Data;
+use chrono::NaiveDateTime;
+
+/// Return where the passed date is (Ok(usize)) or should go (Err(usize)) in
+/// the data slice.
+fn get_insert_location(data: &[Data], date: &NaiveDateTime) -> ::std::result::Result<usize, usize> {
```

Comment:
Given the return type, this should probably be called `find` or something like that